src/loaders/index.ts

```diff
@@ -0,0 +1,57 @@
+import { LoadSpec, loadSpecToString, IVolumeLoader, VolumeDims } from "./IVolumeLoader";
+
+import { OMEZarrLoader } from "./OmeZarrLoader";
+import { JsonImageInfoLoader } from "./JsonImageInfoLoader";
+import { TiffLoader } from "./TiffLoader";
+import VolumeCache from "../VolumeCache";
+
+const enum VolumeFileFormat {
+  ZARR = "zarr",
+  JSON = "json",
+  TIFF = "tiff",
+}
+
+type CreateZarrLoaderOptions = { path: string; cache?: VolumeCache; scene?: number };
+type CreateJsonLoaderOptions = { path: string | string[]; cache?: VolumeCache };
+type CreateTiffLoaderOptions = { path: string };
+
+type CreateLoaderOptionsMap = {
+  [T in VolumeFileFormat]: {
+    [VolumeFileFormat.ZARR]: CreateZarrLoaderOptions;
+    [VolumeFileFormat.JSON]: CreateJsonLoaderOptions;
+    [VolumeFileFormat.TIFF]: CreateTiffLoaderOptions;
+  }[T];
+};
+
+type CreateLoaderOptions = CreateZarrLoaderOptions | CreateJsonLoaderOptions | CreateTiffLoaderOptions;
+
+async function createVolumeLoader<T extends VolumeFileFormat>(
+  fileType: T,
+  options?: CreateLoaderOptionsMap[T]
+): Promise<IVolumeLoader> {
+  switch (fileType) {
+    case VolumeFileFormat.ZARR:
+      const zOptions = options as CreateZarrLoaderOptions;
+      return await OMEZarrLoader.createLoader(zOptions.path, zOptions.scene, zOptions.cache);
```

Comment:
What happens if loader creation fails?  If one of the constructors throws, do we get a rejected promise?

Reply:
Yes - `async` functions transform runtime errors into rejected promises, and `await`ing a rejected promise throws a runtime error.

Reply:
so the place where we `await` this function's results need to be ready to catch things (thinking about w3cv)